addons/web/tests/test_js.py

```diff
@@ -22,12 +23,72 @@ def qunit_error_checker(message):
 
 
 @odoo.tests.tagged('post_install', '-at_install')
-class WebSuite(odoo.tests.HttpCase):
+class WebsuiteCommon(odoo.tests.HttpCase):
+    def get_filter(self, test_params):
+        positive = []
+        negative = []
+        for sign, param in test_params:
+            filters = param.split(',')
+            for filter in filters:
+                filter = filter.strip()
+                if not filter:
+                    continue
+                negate = sign == '-'
+                if filter.startswith('-'):
+                    negate = not negate
+                    filter = filter[1:]
```

Comment:
Isn't enabling negative filters inside params a bit redundant and potentially confusing?
`--test-tags .test_js[-mail]` actually _enables_ `.test_js` but passes a _kind of_ negative parameter, which is equivalent to `--test-tags .test_js,-.test_js[mail]`.
I am just wondering whether this complexity is useful.

Reply:
I thinks that it can be easier to use locally but I'm not sure it is needed, maybe we can check with framework-js

It allows to make something like `.test_js[mail,-mail > Components]` instead of `.test_js[mail].test_js[-mail > Components]`

On runbot, it could allows to disable a whole module except some important tests as an example.
.test_js[mail,-mail_enterprise] would allow to disable mail test but not mail enterprise ones I guess since we don't have a good control over the module in qunit.

 Will we use that? Not likely. Only experience will tell.
 Will framework js use that?  I don't know.

Reply:
The use-case wasn't clear to me, thanks for clarifying.
I suggest to consult the JS team.
I you're not sure about whether it is useful, then keep the simple implementation only (negatives outside parameters only).

Reply:
cc @ged-odoo @aab-odoo